### PR TITLE
Upgrade clang to version 5 in Dockerfile

### PR DESCRIPTION
**Pull request by matt-kwong**
Testing for now

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: templates/tools/dockerfile/clang_update.include [hunk 1]
```diff
@@ -1,28 +1,34 @@
...
+RUN apt-get update && apt-get -y install python && apt-get clean
 
-RUN git clone -n -b release_38 http://llvm.org/git/llvm.git && ${'\\'}
-  cd llvm && git checkout ad57503 && cd ..
-RUN git clone -n -b release_38 http://llvm.org/git/clang.git && ${'\\'}
-  cd clang && git checkout ad2c56e && cd ..
-RUN git clone -n -b release_38 http://llvm.org/git/compiler-rt.git && ${'\\'}
-  cd compiler-rt && git checkout 3176922 && cd ..
-RUN git clone -n -b release_38 ${'\\'}
+RUN git clone -n -b release_50 http://llvm.org/git/llvm.git && ${'\\'}
```
Comment (jtattermusch) [inline templates/tools/dockerfile/clang_update.include]: nit: I am not a fan of building clang from source - it takes long time and brings some unnecessary clutter with it. 

Have we looked for binary distribution of clang?

**Comment by matt-kwong** [comment]
Foundry uses Clang 7, which isn't available as a native package for any Ubuntu version https://packages.ubuntu.com/search?keywords=clang.

I'll see if this can be done purely using binary distributions before merging this.

File: tools/dockerfile/test/fuzzer/Dockerfile [hunk 2]
```diff
@@ -75,28 +75,34 @@ RUN apt-get update && apt-get -y install libgflags-dev libgtest-dev libc++-dev c
...
+RUN git clone -n -b release_50 http://llvm.org/git/libcxxabi.git && \
+  cd libcxxabi && git checkout 5df6b5d && cd ..
 
 RUN mv clang llvm/tools
 RUN mv compiler-rt llvm/projects
 RUN mv clang-tools-extra llvm/tools/clang/tools
 RUN mv libcxx llvm/projects
 RUN mv libcxxabi llvm/projects
 
+# Install newer version of cmake to support building Clang.
```
Comment (jtattermusch) [inline tools/dockerfile/test/fuzzer/Dockerfile]: can we install cmake from a binary package?

**Comment by jtattermusch** [comment]
@matt-kwong   can we use Clang 7 in this PR then?  I think there's value in using the same clang version as foundry

**Comment by matt-kwong** [comment]
@jtattermusch I agree about using the same version as RBE being useful, but they've been having many issues with getting our sanitizers to work. I'll try testing Clang 7 to see if it's more stable outside of RBE.

**Comment by jtattermusch** [comment]
@matt-kwong  I see. If getting clang7 work would be too much hassle, we can stick with 5.0

**Comment by jtattermusch** [comment]
#15481 has been merged. We can now rebase the the change and remove the cmake installation altogether.

**Comment by matt-kwong** [comment]
@jtattermusch I'm thinking of using a pinned version of RBE's Debian8 Clang image as the base for our Dockerimage. Thoughts or objections? It seems like it's pretty bare aside from Clang/sanitizer add-ons, so it should be a suitable base.

**Comment by matt-kwong** [comment]
Closing in favor of #15499

**Pull request closed without merging**